src/snowflake/cli/api/sql_execution.py

```diff
@@ -188,13 +193,21 @@ def show_specific_object(
 
         if show_obj_cursor.rowcount is None:
             raise SnowflakeSQLExecutionError(show_obj_query)
-        elif show_obj_cursor.rowcount > 1:
-            raise ProgrammingError(
-                f"Received multiple rows from result of SQL statement: {show_obj_query}. Usage of 'show_specific_object' may not be properly scoped."
-            )
 
-        show_obj_row = find_first_row(
+        matching_rows = find_all_rows(
             show_obj_cursor,
             lambda row: row[name_col] == unquote_identifier(unqualified_name),
         )
-        return show_obj_row
+
+        if len(matching_rows) == 0:
```

Comment:
Is this condition same as line 194, do we need an explicit check?

Reply:
No, `None` means that the query was not successfully executed, whereas this is looking for a successfully-executed query with a zero-length list.